pkg/beacon/relay/groupselection/groupselection.go

```diff
@@ -56,223 +46,214 @@ func CandidateToNewGroup(
 	if err != nil {
 		return err
 	}
-	initialSubmissionTickets, reactiveSubmissionTickets, err :=
-		generateTickets(
-			newEntry.Bytes(),
-			staker.Address(),
-			availableStake,
-			chainConfig.MinimumStake,
-			naturalThreshold(chainConfig),
-		)
+
+	tickets, err := generateTickets(
+		newEntry.Bytes(),
+		staker.Address(),
+		availableStake,
+		chainConfig.MinimumStake,
+	)
 	if err != nil {
 		return err
 	}
 
-	logger.Infof(
-		"generated [%v] tickets for initial submission phase and [%v] "+
-			"tickets for reactive submission phase",
-		len(initialSubmissionTickets),
-		len(reactiveSubmissionTickets),
-	)
+	logger.Infof("generated [%v] tickets", len(tickets))
 
 	return startTicketSubmission(
-		initialSubmissionTickets,
-		reactiveSubmissionTickets,
+		tickets,
 		relayChain,
 		blockCounter,
 		chainConfig,
 		startBlockHeight,
 		onGroupSelected,
+		ticketSubmissionRoundDuration,
 	)
 }
 
 func startTicketSubmission(
-	initialSubmissionTickets []*ticket,
-	reactiveSubmissionTickets []*ticket,
+	tickets []*ticket,
 	relayChain relaychain.GroupSelectionInterface,
 	blockCounter chain.BlockCounter,
 	chainConfig *config.Chain,
 	startBlockHeight uint64,
 	onGroupSelected func(*Result),
+	ticketSubmissionRoundDuration uint64,
 ) error {
-	initialSubmissionTimeout, err := blockCounter.BlockHeightWaiter(
-		startBlockHeight + chainConfig.TicketSubmissionTimeout/2,
-	)
-	if err != nil {
-		return err
-	}
-
-	reactiveSubmissionTimeout, err := blockCounter.BlockHeightWaiter(
+	ticketSubmissionTimeoutChannel, err := blockCounter.BlockHeightWaiter(
```

Comment:
Why don't we use it as `quitTicketSubmission` and avoid creating additional channel?

Reply:
We must close the channel in order to terminate all submission goroutines not just pass a value which will be received only by one goroutine. I think we should avoid closing `ticketSubmissionTimeoutChannel` because we are on the receiver side here and as a rule of thumb in Go a receiver should never close the channel.

Reply:
The thing is that I don't think we need more than one goroutine now. The reason for triggering `submitTickets` in a separate goroutine is from ancient times. We were submitting a massive number of tickets, not even limited by a group size from what I remember. There is nothing expensive happening in `submitTickets` function, I think we can get rid of `quit <-chan struct{}` parameter from there and just submit tickets in a loop. 

It makes even more sense to do it this way so that we avoid potential nonce duplicate problems when two rounds overlap.

Reply:
Honestly, if we move to just a loop and get rid of a goroutine I am not sure if we need `ticketSubmissionTimeoutChannel` and `closeChannel` in this function at all. At maximum, we'll submit 64 tickets, all in the last round. And we have 3 + 6 = 9 blocks for it. Even for a local environment, this should be more than enough.

Reply:
Ok, I checked if nothing wrong happens when we remove the goroutine and everything seems to be good. Regarding `ticketSubmissionTimeoutChannel` we need it in order to prevent calling `relayChain.GetSelectedParticipants()` too early.  Though, I agree with [this comment](https://github.com/keep-network/keep-core/pull/1510#discussion_r402879191) and I will refactor this in the proposed way.